packages/odf/components/IAM/EditDescriptionTag.tsx

```diff
@@ -0,0 +1,138 @@
+import * as React from 'react';
+import { useCustomTranslation } from '@odf/shared';
+import {
+  Button,
+  Form,
+  FormGroup,
+  FormHelperText,
+  HelperText,
+  HelperTextItem,
+  Modal,
+  ModalVariant,
+  TextInput,
+  Title,
+} from '@patternfly/react-core';
+import { ExclamationCircleIcon, HelpIcon } from '@patternfly/react-icons';
+import { NoobaaIAMContext } from './iam-context';
+import { validateValue } from './validationUtils';
+
+interface EditDescriptionTagProps {
+  name: string;
+  AccessKeyId: string;
+}
+
+const EditDescriptionTag: React.FC<EditDescriptionTagProps> = ({
+  name,
+  AccessKeyId,
+}) => {
+  const { t } = useCustomTranslation();
+  const { noobaaIAM } = React.useContext(NoobaaIAMContext);
+  const [isOpen, setIsOpen] = React.useState<boolean>(true);
+  const descriptiontagRef = React.useRef<HTMLInputElement>(null);
+  const [error, setError] = React.useState<string>('');
+  const [hasAttemptedSubmit, setHasAttemptedSubmit] =
+    React.useState<boolean>(false);
+
+  const handleSubmit = async (
+    event: React.MouseEvent<HTMLButtonElement>
+  ): Promise<void> => {
+    event.preventDefault();
+    setHasAttemptedSubmit(true);
+    const value = descriptiontagRef.current?.value || '';
```

Comment:
why do we need descriptiontagRef ??
```suggestion
    const value = event.target.value || '';
```